Temporarily fix CI by downgrading to `windows-2019`

grtlr opened the pull request:
# Description of change

<!-- Please write a summary of your changes and why you made them. -->
Temporarily fix CI by downgrading to `windows-2019`.

More information: https://github.blog/changelog/2022-01-11-github-actions-jobs-running-on-windows-latest-are-now-running-on-windows-server-2022/

## How the change has been tested

CI runs.

<!-- Describe the tests that you ran to verify your changes. -->

<!-- Make sure to provide instructions for the maintainer as well as any relevant configurations. -->

## Change checklist

<!-- Tick the boxes that are relevant to your changes, and delete any items that are not. -->

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

grtlr committed:
Get rid of `experimental`

grtlr committed:
Automatically cancel older runs

grtlr committed:
Test `jq` on all plattforms

grtlr committed:
Also limit doc builds

grtlr committed:
Revert names

pvdrz reviewed (dismissed):
Maybe we should investigate (in a different PR) running tests with sets of features for each crate to guarantee that all cases are covered.

grtlr committed:
Temporarily switch back to `windows-2019`

More information: https://github.blog/changelog/2022-01-11-github-actions-jobs-running-on-windows-latest-are-now-running-on-windows-server-2022/

grtlr committed:
More robust windows detection

grtlr committed:
Fix

grtlr commented:
@pvdrz I agree, that would be ideal. I have gutted this PR to finally fix the CI for now. I will work on compiling crates separately later on.
